.github/workflows/ci.yml

```diff
@@ -564,26 +564,6 @@ jobs:
       - name: Run the tests
         run: invoke ci-tests -m unit --xdist --slowest=10  --timeout=2.0
 
-  numpy-2:
```

Comment:
I've observed that removing the numpy upperbound allows numpy2 to be installed in our regular unit tests so we no longer need this stage.